### DISCUSSION TITLE
column-level security

### DISCUSSION THREAD

**Discussion by MahmoudElhalwany** [apache/superset#19395] [Ideas]
Many BI applications, particularly in multi-tenancy scenarios, require support for column-level security. That is, the ability to show different columns of a table to users based on some user attribute.

**Comment by srinify** [comment]
Hey there, we like to keep Github Issues focused on bugs! So I'm moving this to Discussions for now :)

**Reply by sandywaves07** [reply]
@MahmoudElhalwany Have you figured out how to handle the column level security? I do have similar use cases, thinking to utilize row-level security features to build column-level security.

**Comment by a-pono** [comment]
Any plans to implement this feature?

**Comment by kky0426** [comment]
Any plans to implement this feature?

**Comment by thefoodiecoder** [comment]
Any plans to implement this? This feature has a lot of importance when it comes to Data Governance. 

We have many database tables where we would like to limit some information from developers/qa/sales roles while allowing them to view all other columns. An example would be the per transaction pricing being charged to a customer, monthly bill amount etc. 

I hope this gets picked up and role based column access is implemented to handle the same.

**Comment by giuliosmall** [comment]
Many Superset/Preset competitors have this feature already available. Is there a plan to add it by any chance?

**Comment by asravankumar** [comment]
Any plans to implement this feature ?

**Comment by littlehera** [comment]
Any plans to implement this feature? This would really be useful when hiding/restricting the data access a user group has. Other BI/Visualization tools already have this.

**Comment by vitaliypet** [comment]
Any plans to implement this feature? This would really be useful when hiding/restricting the data access a user group has. Other BI/Visualization tools already have this.
